brother_ql/backends/helpers.py

```diff
@@ -131,21 +130,143 @@ def status(
     be = backend_factory(selected_backend)
     BrotherQLBackend = be["backend_class"]
     printer = BrotherQLBackend(printer_identifier)
+    return printer
 
-    logger.info("Sending status information request to the printer.")
-    printer.write(b"\x1b\x69\x53")  # "ESC i S" Status information request
+def get_status(
+    printer,
+    receive_only=False,
+    target_status=None,
+):
+    """
+    Get printer status.
+
+    :param BrotherQLBackendGeneric printer: A printer instance.
+    :param bool receive_only: Don't send the status request command.
+    :param int target_status: Expected status code.
+    """
+
+    if not receive_only:
+        printer.write(b"\x1b\x69\x53")  # "ESC i S" Status information request
     data = printer.read()
     try:
         result = interpret_response(data)
     except ValueError:
         logger.error("Failed to parse response data: %s", data)
+    if target_status is not None:
+        if result['status_code'] != target_status:
+            raise ValueError(f"Printer reported 0x{result['status_code']:02x} status instead of 0x{target_status:02x}")
+    return result
+
+
+def get_setting(
+    printer,
+    setting,
+    payload=None
+):
+    """
+    Get setting from printer.
 
-    logger.info(f"Printer Series Code: 0x{result['series_code']:02x}")
-    logger.info(f"Printer Model Code: 0x{result['model_code']:02x}")
-    logger.info(f"Printer Status Type: {result['status_type']} ")
-    logger.info(f"Printer Phase Type: {result['phase_type']})")
-    logger.info(f"Printer Errors: {result['errors']}")
-    logger.info(f"Media Type: {result['media_type']}")
-    logger.info(f"Media Size: {result['media_width']} x {result['media_length']} mm")
+    :param BrotherQLBackendGeneric printer: A printer instance.
+    :param int setting: The code for the setting.
+    :param bytes payload: Optional additional payload, usually not required.
+    """
 
+    # ensure printer is free of errors before proceeding
+    get_status(printer, target_status=0x0)
+    # switch to raster command mode
+    printer.write(b"\x1b\x69\x61\x01")
+    # send command
+    command = b"\x1b\x69\x55" + setting.to_bytes(1) + b"\x01"
```

Comment:
Is it normal to have to enter "raster command mode" before each one of these functions?

In this function you use these new "magic bytes" in a few places. Could we define these magic bytes somewhere else, potentially `brother_ql/reader.py` since that is where we have a lot of other magic bytes defined?

Reply:
Yes, this implementation is based on real USB traffic between the Windows software and the printer, not just trial and error. The magic bytes are documented in both the issue description and in `reader.py` (to a limited degree).

Reply:
Added some more comments for the magic bytes.

Reply:
Thanks for the additional doc. In the future we can maybe make the magic bytes in a variety of places cleaner, but no need to do that here. It all looks good to me!